src/transformers/models/vitpose/image_processing_vitpose.py

```diff
@@ -0,0 +1,649 @@
+                    )
+                    new_images.append(transformed_image)
+            images = new_images
+
+        # since the number of boxes can differ per image, the image processor takes a list
+        # rather than a numpy array of boxes
+        # it currently creates pixel_values of shape (batch_size*num_persons, num_channels, height, width)
+
+        if self.do_rescale:
+            images = [
+                self.rescale(image=image, scale=rescale_factor, input_data_format=input_data_format)
+                for image in images
+            ]
+        if self.do_normalize:
+            images = [
+                self.normalize(image=image, mean=image_mean, std=image_std, input_data_format=input_data_format)
+                for image in images
+            ]
+
+        images = [
+            to_channel_dimension_format(image, data_format, input_channel_dim=input_data_format) for image in images
+        ]
+
+        data = {"pixel_values": images}
+        encoded_inputs = BatchFeature(data=data, tensor_type=return_tensors)
+
+        return encoded_inputs
+
+    def keypoints_from_heatmaps(
+        self,
+        heatmaps,
+        center,
+        scale,
+        kernel=11,
+    ):
+        """
+        Get final keypoint predictions from heatmaps and transform them back to
+        the image.
+
+        Args:
+            heatmaps (`np.ndarray` of shape `(batch_size, num_keypoints, height, width])`):
+                Model predicted heatmaps.
+            center (`np.ndarray` of shape `(batch_size, 2)`):
+                Center of the bounding box (x, y).
+            scale (`np.ndarray` of shape `(batch_size, 2)`):
+                Scale of the bounding box wrt original images of width and height.
+            kernel (int, *optional*, defaults to 11):
+                Gaussian kernel size (K) for modulation, which should match the heatmap gaussian sigma when training.
+                K=17 for sigma=3 and k=11 for sigma=2.
+
+        Returns:
+            tuple: A tuple containing keypoint predictions and scores.
+
+            - preds (`np.ndarray` of shape `(batch_size, num_keypoints, 2)`):
+                Predicted keypoint location in images.
+            - scores (`np.ndarray` of shape `(batch_size, num_keypoints, 1)`):
+                Scores (confidence) of the keypoints.
+        """
+        # Avoid mutation
+        heatmaps = heatmaps.numpy().copy()
+
+        batch_size, _, height, width = heatmaps.shape
+
+        coords, scores = get_keypoint_predictions(heatmaps)
+
+        preds = post_dark_udp(coords, heatmaps, kernel=kernel)
+
+        # Transform back to the image
+        for i in range(batch_size):
+            preds[i] = transform_preds(preds[i], center=center[i], scale=scale[i], output_size=[height, width])
+
+        return preds, scores
+
+    def post_process_pose_estimation(self, outputs, boxes, kernel_size=11):
```

Comment:
Just to clarify, this postprocessing method doesn't take `target_sizes` as input just like the other postprocessing methods?

Reply:
It is little bit debatable since we use the `self.size["width"], self.size["height"]`.

Reply:
We shouldn't use `self.size` directly. Just like in the `preprocess` method, we can default to this is needed, but it should be possible to override. 

As Niels points out, it diverges from other post processing methods -- it would be better to align these and have the method accept `target_sizes`

Reply:
@amyeroberts @NielsRogge I think I have to explain this in details. https://github.com/huggingface/transformers/pull/30530/commits/220859d1eebae98a9f4ddcf47f803d3873950a57

As we all know, normal object detection postprocess pipeline usually requires `target_sizes` for to make normalized output bbox coordinate [0,1] to [0, image_size].

However, in this case we do not have to accept `target_sizes` because we already accept the **image coordinate `boxes`**.

So the overall pipeline is changed as follow.
1. Original Image [H, W]
2. Resized Image [h, w] (we use `self.size` here) e.g. [B, C, 256, 192]
3. Model_run
4. Output Heatmap [h/4, w/4] e.g. ([B, num_keypoints, 64, 48])
5. Postprocess
If the box coordinate is normalized we accept target_sizes to reconstruct the box, otherwise it will not be used.

WDYT?